scripts/maas_inventory_script.py

```diff
@@ -15,47 +15,42 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-import base64
+import argparse
 import json
 import os
-import sys
-import urllib.request
+from urllib.request import urlopen, Request
 
-inventory_url = 'http://hostname:5000/maas/inventory'
+
+def maas_inventory(args):
+    """Fetches the MaaS inventory from the GRNet Ansible Inventory Server
+    and prints it to stdout"""
+
+    params = {
+        'maas': {
+            'url': args.maas_url,
+            'apikey': args.maas_apikey,
+        }
+    }
+
+    url = '{}/maas/inventory'.format(args.server)
+    res = urlopen(
+        Request(url, method='GET'), data=json.dumps(params).encode()).read()
+    print(res.decode())
 
 
 def main():
-    try:
-        # For no arguments, or just --list, just output the inventory.
-        # This allows this script to be used as a dynamic inventory plugin.
-        if not sys.argv[1:] or sys.argv[1] == '--list':
-            # Get API key from AWX credentials
-            apikey = os.getenv('MAAS_APIKEY')
-
-            req = urllib.request.Request(inventory_url)
-            credentials = ':{0}'.format(apikey)
-            encoded_credentials = base64.b64encode(credentials.encode('ascii'))
-            req.add_header(
-                'Authorization',
-                'Basic {}'.format(encoded_credentials.decode('ascii')))
-            res = urllib.request.urlopen(req).read()
-
-            inventory = json.loads(res.decode('utf-8'))
-            print(json.dumps(inventory, indent=4))
-            sys.exit(0)
-
-        elif sys.argv[1] == '--host':
-            # hostvars not supported yet, exit quickly to minimize the lookup
-            # cost
-            print(json.dumps({}))
-            sys.exit(0)
-
-        else:
-            raise Exception(
-                'Unknown argument. Please use either --list or --host')
-    except Exception as e:
-        print('Error: {}'.format(str(e)), file=sys.stderr)
-        sys.exit(1)
+    parser = argparse.ArgumentParser(
```

Comment:
same with the previous script. These files are meant to be called by ansible-inventory binary. Thus, there is no reason to use parse because they are not purposed to be used directly from a user. In case user wants to call the inventory server then he has to call REST api directly.